### ISSUE TITLE
Error launching ROMs on Ubuntu 14.04

### ISSUE THREAD

**Issue by mikeherber** [libretro/reicast-emulator#23]
I have build and tested on Ubuntu 14.04 64bit. Whenever launched, the emulator crashes with 

RetroArch [libretro INFO] :: Verify Failed  : gtruncate(fd,RAM_SIZE + VRAM_SIZE + ARAM_SIZE)==0
   in _nvmem_alloc_mem -> core/hw/mem/_vmem.cpp : 591
 DEBUGBREAK!

free -m returns:

mem:
Total 7905; used 318; free 7587; shared 30; buffers 25; cached 166;
-/+ buffers/cache:
Swap: 
Total 8112; used 0; free 8112

top shows I have plenty of resources available, im using an i7 acer with 8gb ram.

Can anyone shed some light on this issue?

**Comment by mstanislav** [comment]
Same issue here. I've gone back on commits, but didn't have any luck with the error not occurring. I'm on 16.04 64-bit, i5 with 16GB RAM.

**Comment by Veyrdite** [comment]
Do either of you have any reason to think it's Ubuntu related?

Stab in the dark whilst no-one else in around:  I've read in a few places that incompatible/bad BIOS (that perhaps work on other emu's) cause a variety of strange error messages.  A few days back I had an issue with bad BIOS files that gave errors that seemed completely unrelated to the problem.

Do you have access to different set of DC BIOS files that you could try swapping in?  Otherwise: post their md5sums

```
e10c53c2f8b90bab96ead2d368858623  dc_boot.bin
0a93f7940c455905bea6e392dfde92a4  dc_flash.bin
```

**Comment by mstanislav** [comment]
In my case, at least, the md5sums match. I'm able to use vanilla Reicast just fine. I've tried a few avenues (pre built packages, custom builds) and still run into the same issue with the libretro version. Still entirely possible there's some Ubuntu hardening technique or otherwise causing it to break. I may have to dive in deeper and see what strace has to say. Will try to take a look soon and see if anything jumps out. Appreciate the help :)

**Comment by hizzlekizzle** [comment]
I've been able to run it in Ubuntu just fine, though it's been a couple of months since I tried it. Your BIOS images need to be in system/dc

**Comment by mstanislav** [comment]
Yeah BIOS images are being found from output of retroarch. Can you provide details on what version/platform on Ubuntu you're using, where your libretro reicast came from (package? build from github?), etc. Thanks! Will double check on my end, too.

**Comment by hizzlekizzle** [comment]
It was 14.04. I'm on 16.04 now and haven't tried it there yet. I had compiled myself from git.

Someone recently (on the forum, maybe...?) mentioned having issues with an AMD GPU, so there could be something going on with that, as well. What kind of GPU do you have and, if AMD, are you using the proprietary or Mesa driver?

**Comment by mstanislav** [comment]
Intel video chipset. The trick again is that I can use Reicast proper just fine, it's only libretro's version I am having a struggle with, whether built or packages. Absolutely possibly a video problem still, but seems something specific to libretro reicast or at least, my environment + libretro reicast.

**Comment by twinaphex** [comment]
It is 99% of the time you just not having the right bios files set up as the libretro port expects, and also your system directory needs to be writable.

**Comment by mstanislav** [comment]
Looks like my problem was related to the directory not being writable, as noted. I'm always impressed the kinds of errors that can be generated by basic configuration issues :) Hopefully @mikeherber has similar success double checking that. Thanks for your time (and work) @twinaphex. Appreciate the brainstorming, @hizzlekizzle and @Veyrdite.

**Comment by hizzlekizzle** [comment]
Yeah, the writeability bit me when I was first testing it, as well, since I keep my system directory on a read-only network share.
